### PR TITLE
make serve timeout and watch timeout in client handler tunable

### PR DESCRIPTION
**Pull request by yichengq**
for #1551

### CODE REVIEW

**Comment by xiang90** [comment]
@yichengq 
I think we should:
1. make a default timeout for normal key request -> 30 second at most.
2. no default timeout for long pooling watching

I do not want it to be configurable via command line. 

I have talked about this with @barakmich. We decided to make cluster wide configuration (heartbeat, election + this) to be configurable via HTTP API.  

/cc @philips

**Comment by yichengq** [comment]
@xiang90 No default timeout for watching may lead to fds leaking: https://github.com/coreos/etcd/commit/084dcb5596d52a3a1386d59acced24286137e385
I don't know the actual reasons for fds leaking though.
/cc @philips

File: etcdmain/etcd.go [hunk 1]
```diff
@@ -38,6 +40,13 @@ import (
 const (
 	// the owner can make/remove files inside the directory
 	privateDirMode = 0700
+
+	// time to wait for response from EtcdServer requests
+	// 5s for disk and network delay + 10*heartbeat for commit and possible
+	// leader switch
+	defaultServeTimeout = 5*time.Second + 10*(100*time.Millisecond)
```
Comment (xiang90) [inline etcdmain/etcd.go]: i do not think this should be put into etcd main pkg.

**Comment by yichengq** [comment]
to make it configurable, we need to add cluster-wide configuration (http endpoint) first.
i will update timeout value in another PR.

**Pull request closed without merging**